src/main/java/org/jabref/gui/mergeentries/DiffHighlightingEllipsingTextFlow.java

```diff
@@ -0,0 +1,176 @@
+        return true;
+    }
+
+    private boolean fillUntilOverflowing() {
+        while (getHeight() <= getMaxHeight() && getWidth() <= getMaxWidth()) {
+            if (super.getChildren().size() == allChildren.size()) {
+                if (allChildren.size() > 0) {
+                    // all Texts are displayed, let's make sure all chars are as well
+                    Node lastChildAsShown = super.getChildren().get(super.getChildren().size() - 1);
+                    Node lastChild = allChildren.get(allChildren.size() - 1);
+                    if (lastChildAsShown instanceof Text && ((Text) lastChildAsShown).getText().length() < ((Text) lastChild).getText().length()) {
+                        ((Text) lastChildAsShown).setText(((Text) lastChild).getText());
+                    } else {
+                        // nothing to fill the space with
+                        return false;
+                    }
+                }
+            } else {
+                super.getChildren().add(allChildren.get(super.getChildren().size()));
+            }
+            super.autosize();
+        }
+        return true;
+    }
+
+    private boolean ellipseUntilTextFits() {
+        while (getHeight() > getMaxHeight() || getWidth() > getMaxWidth()) {
+            Text lastChildAsShown = (Text) super.getChildren().remove(super.getChildren().size() - 1);
+            while (getEllipsisString().equals(lastChildAsShown.getText()) || "".equals(lastChildAsShown.getText())) {
+                if (super.getChildren().isEmpty()) {
+                    return false;
+                }
+                lastChildAsShown = (Text) super.getChildren().remove(super.getChildren().size() - 1);
+            }
+            Text shortenedChild = new Text(ellipseString(lastChildAsShown.getText()));
+            shortenedChild.getStyleClass().addAll(lastChildAsShown.getStyleClass());
+            super.getChildren().add(shortenedChild);
+            super.autosize();
+        }
+        return true;
+    }
+
+    public void highlightDiff() {
+        allChildren.clear();
+        if (comparisonString.get() != null && !comparisonString.get().equals(fullText)) {
+            final List<Text> highlightedText;
+            switch (diffMode.getValue()) {
+                case PLAIN:
+                    Text text = new Text(fullText);
+                    text.getStyleClass().add("text-unchanged");
+                    highlightedText = List.of(text);
+                    break;
+                case WORD:
+                    highlightedText = DiffHighlighting.generateDiffHighlighting(comparisonString.get(), fullText, " ");
+                    break;
+                case CHARACTER:
+                    highlightedText = DiffHighlighting.generateDiffHighlighting(comparisonString.get(), fullText, "");
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Not implemented " + diffMode.getValue());
+            }
```

Comment:
modern style switch expression ?
`final List<Text> hightlightedText = switch (diffMode.getValue()) ... yield List.of(text); ...` ?